daemon/cmd/daemon_main.go

```diff
@@ -1231,6 +1232,17 @@ func initEnv(cmd *cobra.Command) {
 		log.Fatal("L7 proxy requires iptables rules (--install-iptables-rules=\"true\")")
 	}
 
+	if option.Config.EnableIPSec && option.Config.Tunnel != option.TunnelDisabled {
+		supportsOutputMask, err := ipsec.ProbeXfrmStateOutputMask()
+		fatalLog := log
+		if err != nil {
+			fatalLog = log.WithError(err)
+		}
+		if !supportsOutputMask {
+			fatalLog.Fatal("IPSec with tunneling requires support for xfrm state output masks (Linux 4.19 or later).")
+		}
```

Comment:
Nit: we can further compact this

```suggestion
		if supported, err := ipsec.ProbeXfrmStateOutputMask(); !supported && err != nil  {
			log.WithError(err).Fatal("IPSec with tunneling requires support for xfrm state output masks (Linux 4.19 or later).")
		}
```

Reply:
I had to read Joe's review 3 times to get it and, once I did, I forgot half of it :facepalm: I think this is going to be my last change for the day :zzz:  

I had to change `ProbeXfrmStateOutputMask` again to remove one other case where `(false, nil)` was returned (when the output mask value is incorrect). I then simplified it to just return an error indicating support or lack of support (i.e., no boolean). That simplifies things a bit more.